### DISCUSSION TITLE
Clear process for generating custom voice

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#284] [Archived]
<USER>>>> vocajon<br/> <DATE> [October 15, 2020, 12:48pm]<br/> <br/> Hi,<br/> <br/> Newbie here, so apologies if I'm missing the obvious. I am trying to<br/> achieve the following and think the steps are as follows, but have a few<br/> gaps. Please can someone help me fill in the gaps/answer the inline<br/> questions in Italics? And of course suggest any steps that may be<br/> missing. Maybe this can become a clear quick-start guide for this<br/> particular aim. Thank you.<br/> <br/> Aim: To install Mozilla TTS on a Linux machine, and fine-tune a<br/> pre-trained LJSpeech with a new voice of my own.<br/> <br/> Steps: *slash* <br/> 1)Install cuda (this will allow the nvidia GPU to be used) *slash* <br/> sudo apt-get install cuda<br/> <br/> 2. Install Mozilla TTS using the Simple packaging method detailed<br/> here: *slash* <br/> <https://github.com/mozilla/TTS/wiki/Released-Models> *slash* <br/> Using package: *slash* <br/> <https://github.com/reuben/TTS/releases/download/ljspeech-fwd-attn-pwgan/TTS-0.0.1+92aea2a-py3-none-any.whl><br/> <br/> Check audio synth is working by running it up with: *slash* <br/> python3 -m TTS.server.server *slash* <br/> Open web page to <http://localhost:5002>. Enter some text and check the<br/> wav file is produced and can play OK.<br/> <br/> Check CUDA is working by: *slash* <br/> python3 -m TTS.train *slash* <br/> Before the usage prompt you should see some info about CUDA:<br/> <br/> > Using CUDA: True *slash* <br/> > Number of GPUs: 1 *slash* <br/> > If it's working Using CUDA should be True, and Number of GPUs match<br/> > what is installed in your machine.<br/> <br/> 3. Record a set of wav files in the LJSpeech format - 22050Hz 16-bit<br/> Mono WAV *slash* <br/> Recommended duration:Single sentences 5 - 10 seconds each *slash* <br/> Remove any silence at the beginning and end of each recording. *slash* <br/> Normalize the audio level. *slash* <br/> Minimum number of files/combined duration: *Help please*<br/> <br/> 4)Store the wavs in a dataset folder with the following structure: *slash* <br/> *slash* |- yourchosenname *slash* <br/> *slash* |- metadata.csv *slash* <br/> *slash* |- wavs *slash* <br/> *slash* |- xxxx-0001.wav *slash* <br/> *slash* |- xxxx-0002.wav<br/> <br/> 5. Create metadata.csv inside your dataset folder with the following<br/> format: *slash* <br/> xxxx-0001 *slash* |There were 50 people in the room. *slash* |There were fifty<br/> people in the room. *slash* <br/> xxxx-0002 *slash* |Mr Jones is the friendly local butcher. *slash* |Mister Jones is<br/> the friendly local butcher. *slash* <br/> Using pipes to separate the 3 columns. *slash* <br/> Where the 3rd column expands numbers, titles etc.<br/> <br/> *Help please - is it necessary to split this into metadata_train.csv and<br/> metadata_val.csv? I saw suggestions val should be about 10% of the size<br/> of train?*<br/> <br/> 6. Pre-process *Please Help - I think there should be some kind of<br/> preprocess stage here? But don't see any .py for doing this in this<br/> repo? Is it done automatically as part of the training? I'm guessing<br/> so as there are params about do_trim_silence etc.*<br/> <br/> 7. Prepare a config.json for your new dataset *slash* <br/> *Please help - I am not sure what needs changing in here. I'm<br/> thinking the following?: *slash* <br/> restore_path - set to the new dataset? But I'm not sure how to get<br/> the dataset into .pth.rar format? *slash* <br/> run_name - set to new dataset name (although probably not<br/> mandatory?) *slash* <br/> run_description - describe new dataset *slash* <br/> mel_fmin - set to *slash* ~50 for male, *slash* ~95 for female *slash* <br/> batch_size - 32 standard. I understand there are issues with GPU's<br/> with smaller amounts of memory, and that you really need 16GB. Are<br/> there any recommendations (e.g. drop this value) if trying to use a<br/> 4-8GB GPU? Or are you just wasting your time? *slash* <br/> output_path - do these need changing? *slash* <br/> datasets - set the name and path to match your new dataset path?*<br/> <br/> 8. Fine tune the model: *slash* <br/> python3 -m TTS.train *slash* --config_path TTS/tts/configs/config.json<br/> *slash* --restore_path /path/to/your/model.pth.tar<br/> <br/> 9. Run up the server as in step 2 and test the new voice.<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/clear-process-for-generating-custom-voice]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [October 25, 2020, 1:13am]<br/> <br/> couple notes;<br/> <br/> for step 1 it is way easier to use `conda`. You don't manually deal with<br/> cuda which might be a problem.<br/> <br/> you also need to find the right audio values for your dataset like<br/> silence threshold, normalization, etc. Our data analysis notebook can<br/> help. *slash* <br/> <br/> ![](https://github.githubassets.com/favicons/favicon.svg){.site-icon<br/> [GitHub](https://github.com/mozilla/TTS/tree/master/notebooks/dataset_analysis)<br/> <br/> ### [mozilla/TTS](https://github.com/mozilla/TTS/tree/master/notebooks/dataset_analysis)<br/> <br/> forum: https://discourse.mozilla.org/c/tts) - mozilla/TTS<br/> <br/> You should also remove noisy samples and do some running based on<br/> quality. Again you can use CheckSNR notebook for this to start.<br/> <br/> Also if you create your own dataset, you should also perform a phoneme<br/> coverage filtering to create your transcript set as representative as<br/> possible for the target language.<br/> <br/> BTW, if there is any volunteer to create a nice script or notebook to<br/> automate these steps to make life easier for beginners, we can work on<br/> that together.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> rdh<br/> <DATE> [October 30, 2020, 11:48am]<br/> <br/> > Minimum number of files/combined duration: *Help please*<br/> <br/> I've been wanting to find the answer to this myself.<br/> <br/> In [my<br/> experience](https://drive.google.com/drive/folders/1SkDGmla3Ohiiaaah_syK8U3S1GsHrBq1)<br/> for Dutch, 4.000 recordings were not sufficient for transfer learning<br/> using Tacotron DDC, using near to default configuration.<br/> <br/> For completeness, I started off from [this<br/> model](https://drive.google.com/drive/folders/1Ew_UGWNNqYxim8H4hhTjeApMRUAcEHZz),<br/> which was trained with 15.000 fragments.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [October 30, 2020, 1:05pm]<br/> <br/> can you share a couple of samples from your dataset to see the quality?<br/> I can comment better.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> rdh<br/> <DATE> [October 30, 2020, 1:43pm]<br/> <br/> I [appended some<br/> samples](https://drive.google.com/drive/folders/1SkDGmla3Ohiiaaah_syK8U3S1GsHrBq1)<br/> of the second dataset. Samples from the original dataset can be found on<br/> my [dataset repo](https://github.com/r-dh/dutch-vl-tts).<br/> <br/> [Archived Post]<br/>
